app/validators/ancestry_depth_validator.rb

```diff
@@ -12,9 +16,10 @@
 validates :parent, ancestry_depth: { minimum: 1, maximum: 3 }
 
 =end
-
 class AncestryDepthValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
+    return if record.errors.include? attribute
```

Comment:
Can you be sure that the other validator will be run first? I have no idea what the validation order is.